cmake/modules/shields.cmake

```
@@ -79,41 +79,65 @@ foreach(root ${BOARD_ROOT})
 
       list(REMOVE_ITEM SHIELD-NOTFOUND ${s})
 
-      # Add <shield>.overlay to the shield_dts_files output variable.
+      # Add <shield>.overlay to a temporary list
       list(APPEND
-        shield_dts_files
+        shield_${s}_dts_files
         ${SHIELD_DIR_${s}}/${s}.overlay
         )
```

Minor.

Each shield can only have a single overlay file, hence there is no reason for a list in this case.
And as the variable now is containing the shield name, it should just be singular:
```suggestion
      set(shield_${s}_dts_file ${SHIELD_DIR_${s}}/${s}.overlay)
```

---

Fixed